### PR TITLE
set up jacoco for code coverage

### PR DESCRIPTION
**Pull request by yu-sun-77**
*Fixes #, if available:* #236 

*Description of changes:*

1. add jacoco to set up code coverage.
2. show code coverage report for every pull request in future.
3. integrate with "codecov" to show latest code coverage badge in REAMDE file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yojs** [comment]
@yu-sun-77 can we create an issue and add it in the `Fixes #, if available:` section.
And also a description, what this change does

**Review by yojs — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -122,6 +123,44 @@ tasks.withType(JavaCompile) {
...
+        csv.enabled false
+        html.destination file("${buildDir}/jacocoHtml")
+    }
+}
+
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            limit {
+                minimum = 0.8
```
[the remark this answers is not in the document]
Comment (yu-sun-77) [reply build.gradle]: It seems you're referring to outdated version of code, I've changed to 0.0 for now. Can you please refresh and review again? Thanks.

File: build.gradle [hunk 2]
```diff
@@ -122,6 +123,44 @@ tasks.withType(JavaCompile) {
     options.warnings = false
 }
 
+jacoco {
+    toolVersion = "0.8.5"
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled false
```
[the remark this answers is not in the document]
Comment (yu-sun-77) [reply build.gradle]: Latest code has been changed to `true`. Can you please refresh and review again? Thanks.

**Comment by yu-sun-77** [comment]
> @yu-sun-77 can we create an issue and add it in the `Fixes #, if available:` section.
> And also a description, what this change does

I've created an issue and update description. thanks!

**Review by rguo-aws — APPROVED** [review]
LGTM

**Pull request merged**